File: src/main/java/com/lanka_guide/tripplanner/model/activity/transport/Transport.java
package com.lanka_guide.tripplanner.model.activity.transport;

import com.lanka_guide.tripplanner.model.Location;
import com.lanka_guide.tripplanner.model.activity.Activity;

import java.time.ZonedDateTime;
import java.util.List;

public class Transport extends Activity {
    Location startLocation;
    Location endLocation;

    public Transport(ZonedDateTime from, ZonedDateTime to) {
        super(from, to);
    }
}

File: src/main/kotlin/com/lankaguide/tripplanner/model/activity/transport/Transport.kt
package com.lankaguide.tripplanner.model.activity.transport

import com.lankaguide.tripplanner.model.activity.Activity
import com.lankaguide.tripplanner.model.place.Place

import java.time.ZonedDateTime

abstract class Transport(open val from: Place, open val to: Place, override val start: ZonedDateTime, override val end: ZonedDateTime) : Activity(start, end)